apps/web/components/eventtype/EventAdvancedTab.tsx

```diff
@@ -386,20 +403,22 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupInfered
                 selectedCustomInput={selectedCustomInput}
                 onSubmit={(values) => {
                   const customInput: CustomInputParsed = {
-                    id: -1,
+                    id: getRandomId(),
                     eventTypeId: -1,
                     label: values.label,
                     placeholder: values.placeholder,
                     required: values.required,
                     type: values.type,
                     options: values.options,
+                    hasToBeCreated: true,
```

Comment:
this variable would be used in handleCustomInputs(https://github.com/calcom/cal.com/blob/main/packages/trpc/server/routers/viewer/eventTypes.tsx#L38) to detect if we have to create a custom input. we can also choose to not make this change this but this makes it more obvious that we are creating custom input.